Campaign Wizard: Missing required field: Start Date

<!--- Provide a general summary of the issue in the **Title** above -->
<!--- Before you open an issue, please check if a similar issue already exists or has been closed before. --->

This has been reported before by a number of people including me.
 - https://suitecrm.com/forum/suitecrm-7-0-discussion/11680-email-marketing-start-date-and-time
 - https://suitecrm.com/forum/suitecrm-7-0-discussion/13372-missing-required-field-start-date
google "suitecrm campaign missing start date" to find others.

When using the campaign wizard to create and launch a campaign you are required to use the "next button" to continue, this will fail at "Marketing" as the date CANNOT BE SET - it refuses to accept any date.

It is a REAL hassle as people cannot send campaigns.

#### Issue
<!--- Provide a more detailed introduction to the issue itself, and why you consider it to be a bug -->

I have set the correct and required information in php.ini:
 date.timezone="Australia/Melbourne"

I have also set in the admin panel the following information:
 date format: 23/12/2006
 time format: 23:00

I have also set the same information for the users dealing with campaigns.

Yet there is a message in the LOG file for SUITECRM
Tue Jun 13 11:31:36 2017 [11137][1][ERROR] convert: Conversion of : 00.00 from d/m/Y H.i to Y-m-d H:i:s failed
Tue Jun 13 11:31:54 2017 [11137][1][ERROR] convert: Conversion of 13/06/2017 11:00 from d/m/Y H.i to Y-m-d H:i:s failed

Note that the time for the log message "Tue Jun 13 11:31:36 2017" is the exact time and date when it happened.

#### Expected Behavior
<!--- Tell us what should happen -->

The wizard should accept the date for the SELECTED time and date.
We do not enter the time by hand but use the provided calendar.

#### Actual Behavior
<!--- Tell us what happens instead -->

The wizard DOES NOT accept the date for the SELECTED time and date.
We do not enter the time by hand but use the provided calendar.

#### Possible Fix
<!--- Not obligatory, but suggest a fix or reason for the bug -->

I checked where this error happens but do not have enough time to fix this, further I do not 100% understand the DATE TIME classes of SuiteCRM.

The error is created in the file "SUITECRMINSTALL/include/TimeDate.php" on line 963, alternatively search for "convert:" in the file. 

The correct time zones are passed to the function (UTC, Australia/Melbourne), the correct time is passed. Everything seems alright but I cannot get passed this function.

#### Steps to Reproduce
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug include code to reproduce, if relevant -->
There are many possible ways to get to this bug. Once possible way is to EDIT the campaign again to actually send a test email, you cannot go past "Marketing", it does not accept the date.
1. create a campaign
2. create a template
3. go back to view campaigns 
4. select/edit campaign
5. click "next" to go to Marketing
6. enter a date, marketing email name and other required fields
7. it will come back to the same Marketing, with ALL fields filled out BUT THE DATE
it will display error: Missing required field: Start Date

#### Context
<!--- How has this bug affected you? What were you trying to accomplish? -->
<!--- If you feel this should be a low/medium/high priority then please state so -->

People cannot send campaigns without me filling out the missing dates in the DATABASE.
I need to DIRECTLY enter the dates into the database - but even that makes it fail half the time.

To actually SEND the campaign I need to bypass everything I need to do the following
 - go to list campaigns
 - click on the campaing
 - on the button left top corner click on "send test" or "send emails"

#### Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->
* SuiteCRM Version used:
* Browser name and version (e.g. Chrome Version 51.0.2704.63 (64-bit)): DOES NOT MATTER
* Environment name and version (e.g. MySQL, PHP 7): 5.6.30 and 7.2.2
* Operating System and version (e.g Ubuntu 16.04): CentOS

I think this should be resolved in #4702

According to the error message from the OP

Tue Jun 13 11:31:54 2017 [11137][1][ERROR] convert: Conversion of 13/06/2017 11:00 from d/m/Y H.i to Y-m-d H:i:s failed

The expected format in the convert function was d/m/Y H.i but the time string supplied was 13/06/2017 11:00.

After the changes I made I have not been able to reproduce this error anymore.

--- a/modules/Campaigns/WizardMarketing.html
+++ b/modules/Campaigns/WizardMarketing.html
@@ -880,6 +880,7 @@
 					<table border="0" cellpadding="0" cellspacing="0" class="dateTime">
 						<tr valign="middle">
 							<td nowrap class="datafield" style="width:150px;">
+								<input type="hidden" id="userTimeFormat" value="{$TIME_FORMAT}">
 								<input autocomplete="off" type="text" id="{$fields.date_start.name}_date" value="{$MRKT_DATE_START}" xxx="{$fields[$fields.date_start.name].value}" size="11" maxlength="10" title='' tabindex="0" onblur="combo_{$fields.date_start.name}.update(); parseDate(this, '{$CALENDAR_DATEFORMAT}');" onchange="combo_{$fields.date_start.name}.update();parseDate(this, '{$CALENDAR_DATEFORMAT}'); "    >
 								{capture assign="other_attributes"}alt="{$APP.LBL_ENTER_DATE}" style="position:relative; top:6px" border="0" id="{$fields.date_start.name}_trigger"{/capture}
 								{sugar_getimage name="jscalendar" ext=".gif" other_attributes="$other_attributes"}&nbsp;
@@ -889,7 +890,7 @@
 								<script type="text/javascript">
 									setInterval(function(){
 										$('input[name="wiz_step3_date_start"]').val($('#date_start_date').val());
-										$('input[name="wiz_step3_time_start"]').val($('#date_start_hours').val() + ':' + $('#date_start_minutes').val());
+										$('input[name="wiz_step3_time_start"]').val($('#date_start_hours').val() + $('#userTimeFormat').val()[3] + $('#date_start_minutes').val());
 										$('select[name="meridiem"]').val($('#date_start_meridiem').val());
 										if(parseInt($('#date_start_hours').val())>12) {
 											$('#date_start_meridiem').val('PM');
@@ -926,7 +927,7 @@
 					<script type="text/javascript">
 						var combo_{$fields.date_start.name} = new Datetimecombo("{$fields[$fields.date_start.name].value}", "{$fields.date_start.name}", "{$TIME_FORMAT}", "0", '1', false, true);
 						//Render the remaining widget fields
-						combo_{$fields.date_start.name}.timeseparator = ':';
+						combo_{$fields.date_start.name}.timeseparator = $('#userTimeFormat').val()[3];
 						text = combo_{$fields.date_start.name}.html('');
 						document.getElementById('{$fields.date_start.name}_time_section').innerHTML = text;
 

